api/src/main/resources/marquez/db/migration/V68__alter_datasets_view_to_keep_symlinks.sql

```diff
@@ -0,0 +1,33 @@
+DROP VIEW IF EXISTS datasets_view;
+CREATE VIEW datasets_view AS
+SELECT d.uuid,
+    d.type,
+    d.created_at,
+    d.updated_at,
+    CASE
+    	 WHEN (d.namespace_name = namespaces.name AND d.name = symlinks.name) THEN d.namespace_uuid
+    	 ELSE namespaces.uuid
+    END
+    AS namespace_uuid ,
+    d.source_uuid,
+    CASE
+    	 WHEN (d.namespace_name = namespaces.name and d.name = symlinks.name) THEN d.name
+    	 ELSE symlinks.name
+    END
+    AS name,
+    array(SELECT ROW(namespaces.name::character varying(255), symlinks.name::character varying(255))::dataset_name) AS dataset_symlinks,
+    d.physical_name,
+    d.description,
+    d.current_version_uuid,
+    d.last_modified_at,
+    CASE
+    	 WHEN (d.namespace_name = namespaces.name AND d.name = symlinks.name) THEN d.namespace_name
+    	 ELSE namespaces.name
+    END
+    AS namespace_name,
+    d.source_name,
+    d.is_deleted
+   FROM datasets d
+     JOIN dataset_symlinks symlinks ON d.uuid = symlinks.dataset_uuid
+     JOIN namespaces ON symlinks.namespace_uuid = namespaces.uuid
```

Comment:
The query is not much different from before, but now the dataset uuid is not the primary key anymore since a dataset and his symlink has the same dataset uuid (that's why the group by is not here anymore).
We join this view with dataset_symlinks to identify if it's a primary dataset or not. if it is a primary dataset, the values in the row remain the same as before. If not, the namespace, name and namespace uuid are replaced by the one from the symlinks (value from the join table).

Reply:
Valid, since we recreate the `dataset_view` on every marquez deploy, we can make your changes in [`R__3_Datasets_view.sql`](https://github.com/MarquezProject/marquez/blob/main/api/src/main/resources/marquez/db/migration/R__3_Datasets_view.sql) as @pawel-big-lebowski suggested.